### PR TITLE
lantiq: move dsl related base-files into own package

### PR DESCRIPTION
**Pull request by sch-m**
It does not make sense to install this components on lantiq systems
where the dsl subsystem is not needed/used.

This also makes it possible to use the files also on other targets.
(hopefully ipq401x / FritzBox 7530 in the near future)

This commit is a part of "VRX518 support on ipq40xx" work you can find here:
https://github.com/3headeddevs/openwrt/tree/vrx518_support

Signed-off-by: Martin Schiller <ms.3headeddevs@gmail.com>

### CODE REVIEW

File: package/network/utils/ltq-dsl-base-files/Makefile [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (C) 2019 OpenWrt.org
```
Comment (adrianschmutzler) [inline package/network/utils/ltq-dsl-base-files/Makefile:2]: You are not affiliated with OpenWrt, so AFAIK you cannot put an OpenWrt copyright here.

File: package/network/utils/ltq-dsl-base-files/Makefile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=ltq-dsl-base-files
```
Comment (adrianschmutzler) [inline package/network/utils/ltq-dsl-base-files/Makefile:10]: I'd prefer just "ltq-dsl-base"

**Comment by adrianschmutzler** [comment]
I like this; one could think about renaming the files more appropriately, but I would _not_ do this in this first step, but, if at all, separately afterwards.

File: package/network/utils/ltq-dsl-base-files/Makefile (hunk 1 above)
Comment (sch-m) [reply package/network/utils/ltq-dsl-base-files/Makefile:2]: OK, I'll remove that line.

File: package/network/utils/ltq-dsl-base-files/Makefile (hunk 2 above)
Comment (sch-m) [reply package/network/utils/ltq-dsl-base-files/Makefile:10]: I have chosen the name in reference to the origin of the files it contains, namely /target/linux/lantiq/base-files.

Comment (adrianschmutzler) [reply package/network/utils/ltq-dsl-base-files/Makefile:10]: It's nitpicking, but essentially this is about what this package is supposed to evolve into:

ltq-dsl-base: This means it's the "base package" for the other two packages, and may contain whatever these packages would reasonably share (e.g. one could also move dependencies, setting kernel symbols etc.)

ltq-dsl-base-files: This means the package is limited to containing "files", i.e. shared scripts and library files.

Of course, this is nitpicking, and most probably nobody except me will really care about that :-)

File: package/network/utils/ltq-dsl-base-files/Makefile (hunk 1 above)
Comment (adrianschmutzler) [reply package/network/utils/ltq-dsl-base-files/Makefile:2]: Either put your name there or just omit the copyright and just keep the license.

Comment (sch-m) [reply package/network/utils/ltq-dsl-base-files/Makefile]: Done

File: package/network/utils/ltq-dsl-base-files/Makefile (hunk 2 above)
Comment (sch-m) [reply package/network/utils/ltq-dsl-base-files/Makefile]: OK, I agree.

Comment (adrianschmutzler) [reply package/network/utils/ltq-dsl-base-files/Makefile]: Now, after I've finally offered you the option to choose :-)

**Comment by adrianschmutzler** [comment]
I plan to merge this, I will just leave it open for a few days to give other people the chance to comment/NAK.

**Comment by adrianschmutzler** [comment]
BTW, I'm not really sure whether this should be in "utils" or in "config" subfolder like the other packages, but I don't really have much experience there. Maybe somebody can comment on that.

File: package/network/utils/ltq-dsl-base-files/Makefile (hunk 2 above)
Comment (sch-m) [reply package/network/utils/ltq-dsl-base-files/Makefile]: I agree with you because I like the "not just files" argument.

**Comment by adrianschmutzler** [comment]
> . /usr/share/libubox/jshn.sh

Though it won't matter much practically, the package should add a dependency for that file. Think it would just be `+jshn`

**Comment by adrianschmutzler** [comment]
> [ -f /lib/functions/lantiq_dsl.sh ] && . /lib/functions/lantiq_dsl.sh

The -f check can be removed for both packages as we have this file in ltq-dsl-base and that one is selected.

Would you provide that in a second patch and just push it to this PR? (Otherwise I'd do so when merging this ...)

https://github.com/openwrt/openwrt/blob/master/package/network/config/ltq-adsl-app/files/dsl_control#L11
https://github.com/openwrt/openwrt/blob/master/package/network/config/ltq-vdsl-app/files/dsl_control#L11

**Comment by sch-m** [comment]
> > . /usr/share/libubox/jshn.sh
> 
> Though it won't matter much practically, the package should add a dependency for that file. Think it would just be `+jshn`

OK, I'll add this dependency.

**Comment by sch-m** [comment]
> > [ -f /lib/functions/lantiq_dsl.sh ] && . /lib/functions/lantiq_dsl.sh
> 
> The -f check can be removed for both packages as we have this file in ltq-dsl-base and that one is selected.

OK, I'll do so, but this check would not have been necessary in the past either, because the script "lantiq_dsl.sh"
was part of TARGET_LANTIG on which these packages also depend.

**Comment by adrianschmutzler** [comment]
Thanks, I've just merged the PR.

@feckert You should update your patch https://patchwork.ozlabs.org/project/openwrt/patch/20200616082613.892-1-fe@dev.tdt.de/ now. Note that I won't merge that one lacking the background to properly review it, I just commented on the ids back then based on "common sense". But thanks for pushing this one here forward.

**Pull request closed without merging**